draft-wang-ppm-differential-privacy.md

```diff
@@ -414,23 +459,204 @@ random seed into a sample from a given distribution.
 
 > TODO: Specify a Gaussian sampler
 
-## Randomized Response
+## Randomized Response {#rr}
 
 > TODO: Specify any primitives required for randomized response mechanisms
 
 # DP Mechanisms for VDAFs {#mechanisms}
 
 The section defines a generic interface for DP mechanisms VDAFs.
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
-
-## Client-DP for Prio3Histogram
-
-> TODO: Concretely describe a Client-DP mechanism for Prio3Histogram.
-
-## Aggregator-DP for Prio3Histogram
-
-> TODO: Concretely describe an Aggregator-DP mechanism for Prio3Histogram.
+We will define an interface `DpMechanism` that composes the following:
+
+* An optional Client-DP mechanism that adds noise to Clients' measurements.
+
+* An optional Aggregator-DP mechanism that adds noise to an Aggregator's
+  aggregate share, based on the number of measurements, and the minimum
+  batch size.
+
+* An optional debiasing step that removes the bias in DP primitives (i.e.
+  `DpPrimitive.debias`).
+
+The composition of Client- and Aggregator-DP mechanisms enforces the DP
+guarantee of a data collection task.
+
+~~~
+class DpMechanism:
+    Measurement
+    AggregateShare
+    AggregateResult
+    DebiasedAggregateResult
```

Comment:
nit: I think it's safe to assume that AggregateResult and DebiasedAggregateResult are always the same.

Reply:
One difference is when the aggregate result is [integer] and DebiasedAggregateResult is [float]

Reply:
Oh, OK! Couldn't the aggreagte result also be a float?

Reply:
Currently `Prio3FixedPointBoundedL2VecSum` uses aggregate result as an array of float, so that could be the case. And there is nothing needed in debiasing.

Reply:
I suggest having one type for now until we need the flexibility.

Reply:
I think `AggregateResult` is a term defined in VDAF draft. For `Prio3SumVec`, it is an array of unsigned integers. If RR is used, Collector needs to debias aggregate result, which will turn it from an array of unsigned integers to an array of floats. It seems like the distinction is needed here?

Reply:
Why should that be the case? If what we want is vector of sums, then we'd expect integers wouldn't we? Why convert them to floats?

Reply:
Because the arithmetic operations in debiasing results in floats, see Appendix A.2 in https://arxiv.org/pdf/2211.10082.pdf, and also could result in negative values. So for Prio3SumVec for example, aggregate result is an array of *unsigned* integers, debiased aggregate result needs to be at least an array of *signed* integers, or floats.

Reply:
Ack. OK, let's kep it then